R/pkg/R/functions.R

```diff
@@ -2836,7 +2845,11 @@ setMethod("lpad", signature(x = "Column", len = "numeric", pad = "character"),
 setMethod("rand", signature(seed = "missing"),
           function(seed) {
             jc <- callJStatic("org.apache.spark.sql.functions", "rand")
-            column(jc)
+
+            # By assigning a one-row data.frame, the result of this function can be collected
+            # returning a one-element Column
+            df <- as.DataFrame(sparkRSQL.init(), data.frame(0))
```

Comment:
in fact, it might be useful to have a shared empty DataFrame instead of instantiating a new one every time a SQL function is called (which could be quite frequent).